src/Sonarr.Api.V3/Indexers/ReleasePushModule.cs

```diff
@@ -1,50 +1,111 @@
+using NzbDrone.Core.Indexers;
 using NzbDrone.Core.Parser.Model;
 using Sonarr.Http.Extensions;
+using Sonarr.Http.REST;
 
 namespace Sonarr.Api.V3.Indexers
 {
     class ReleasePushModule : ReleaseModuleBase
     {
         private readonly IMakeDownloadDecision _downloadDecisionMaker;
         private readonly IProcessDownloadDecisions _downloadDecisionProcessor;
+        private readonly IIndexerFactory _indexerFactory;
         private readonly Logger _logger;
+        private ResourceValidator<ReleaseResource> _releaseValidator;
 
         public ReleasePushModule(IMakeDownloadDecision downloadDecisionMaker,
                                  IProcessDownloadDecisions downloadDecisionProcessor,
+                                 IIndexerFactory indexerFactory,
                                  Logger logger)
         {
             _downloadDecisionMaker = downloadDecisionMaker;
             _downloadDecisionProcessor = downloadDecisionProcessor;
+            _indexerFactory = indexerFactory;
             _logger = logger;
+;
+            _releaseValidator = new ResourceValidator<ReleaseResource>();
+            _releaseValidator.RuleFor(s => s.Title).NotEmpty();
+            _releaseValidator.RuleFor(s => s.DownloadUrl).NotEmpty();
+            _releaseValidator.RuleFor(s => s.DownloadProtocol).NotEmpty();
+            _releaseValidator.RuleFor(s => s.PublishDate).NotEmpty();
 
             Post["/push"] = x => ProcessRelease(this.Bind<ReleaseResource>());
-
-            PostValidator.RuleFor(s => s.Title).NotEmpty();
-            PostValidator.RuleFor(s => s.DownloadUrl).NotEmpty();
-            PostValidator.RuleFor(s => s.DownloadProtocol).NotEmpty();
-            PostValidator.RuleFor(s => s.PublishDate).NotEmpty();
         }
 
         private Response ProcessRelease(ReleaseResource release)
         {
+            var validationFailures = _releaseValidator.Validate(release).Errors;
+
+            if (validationFailures.Any())
+            {
+                throw new ValidationException(validationFailures);
+            }
+
             _logger.Info("Release pushed: {0} - {1}", release.Title, release.DownloadUrl);
 
             var info = release.ToModel();
 
             info.Guid = "PUSH-" + info.DownloadUrl;
 
+            ResolveIndexer(info);
+
             var decisions = _downloadDecisionMaker.GetRssDecision(new List<ReleaseInfo> { info });
             _downloadDecisionProcessor.ProcessDecisions(decisions);
 
-            return MapDecisions(decisions).First().AsResponse();
+            var firstDecision = decisions.FirstOrDefault();
+
+            if (firstDecision?.RemoteEpisode.ParsedEpisodeInfo == null)
+            {
+                throw new ValidationException(new List<ValidationFailure>{ new ValidationFailure("Title", "Unable to parse", release.Title) });
+            }
+
+            return MapDecisions(new [] { firstDecision }).AsResponse();
+        }
+
+        private void ResolveIndexer(ReleaseInfo release)
```

Comment:
Maybe include this change in the v2 api implementation? Given that third-parties will likely use that api.